### PR TITLE
Migrate JFace Viewer tests to JUnit 4

### PR DESCRIPTION
**Pull request by HeikoKlare**
* Migrates the tests in `org.eclipse.jface.tests.viewer` to JUnit 4
* Replaces some test method precondition checks with assume calls
* Replaces test name access with `TestName` rule

### CODE REVIEW

File: tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/labelProviders/DecoratingStyledCellLabelProviderTest.java [hunk 1]
```diff
@@ -275,6 +292,7 @@ public void testGetBackground() {
 		// getDecoratingStyledLabelProvider().getBackground(ti));
 	}
 
+	@Test
 	public void testGetFont() {
```
Comment (fedejeanne) [inline tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/labelProviders/DecoratingStyledCellLabelProviderTest.java:296]: Could be removed in another PR

File: tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/viewers/AbstractTreeViewerTest.java [hunk 2]
```diff
@@ -229,6 +247,7 @@ public void testRefreshWithDuplicateChild() {
 	 * expanded, B is not - A gets deleted - B gets expanded because it reused A's
 	 * item
 	 */
+	@Test
 	public void testRefreshWithReusedItems() {
```
Comment (fedejeanne) [inline tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/viewers/AbstractTreeViewerTest.java:251]: Could be removed in another PR

File: tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/viewers/AbstractTreeViewerTest.java [hunk 3]
```diff
@@ -272,6 +293,7 @@ public void testSetExpandedWithCycle() {
 	 * Test for Bug 41710 - assertion that an object may not be added to a given
 	 * TreeItem more than once.
 	 */
+	@Test
 	public void testSetDuplicateChild() {
 		// Widget root = fViewer.testFindItem(fRootElement);
 		// assertNotNull(root);
```
Comment (fedejeanne) [inline tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/viewers/AbstractTreeViewerTest.java:299]: Remove

**Review by fedejeanne — APPROVED** [review]

**Review by fedejeanne — APPROVED** [review]

**Comment by vogella** [comment]
Error failure documented here https://github.com/eclipse-platform/eclipse.platform.ui/issues/1005

**Comment by vogella** [comment]
Thanks @HeikoKlare
